internal/prober/scaler/scale.go

```diff
@@ -162,6 +164,16 @@ func (r *resScaler) updateResourceAndScale(ctx context.Context, scaleSubRes *aut
 	}
 	if _, err = r.scaler.Update(childCtx, *gr, scaleSubRes, metav1.UpdateOptions{}); err != nil {
 		return err
+	} else {
```

Comment:
There is no need for `else`. If there is an error during update of `Scale` subresource then the error is returned. IF there is no error then annotation is updated if the operation is scaleUp.

So this can be simplified for better code readability to:
```go
if _, err = r.scaler.Update(childCtx, *gr, scaleSubRes, metav1.UpdateOptions{}); err != nil {
  return err
}
if r.resourceInfo.operation == scaleUp {
  // rest of the code
}

```

Reply:
Done with https://github.com/gardener/dependency-watchdog/pull/140/commits/0909963107f75efcd9bfa5d55df2907023df39a7, that was my bad.